pkg/controller/dataimportcron-controller.go

```diff
@@ -253,6 +263,8 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 
 	desiredDigest := dataImportCron.Annotations[AnnSourceDesiredDigest]
 	digestUpdated := desiredDigest != "" && (len(imports) == 0 || desiredDigest != imports[0].Digest)
+	incrementMetric := true
+	currentCronCopy := dataImportCron.DeepCopy()
```

Comment:
@arnongilboa To prevent a situation where I increment/decrement but fail r.client.Update() as we discussed
Tagging it here so the change doesn't slip away unreviewed

Reply:
all you are interested is in the current `upToDateCondition.ConditionState.Status`, so why DeepCopy is needed?

Reply:
also need to check `upToDateCondition != nil`

Reply:
Done